src/Integration.Vsix/CFamily/RulesMetadataCache.cs

```diff
@@ -36,27 +36,68 @@ internal interface IRulesConfiguration
         IDictionary<string, RuleMetadata> RulesMetadata { get; }
     }
 
-    internal sealed class RulesMetadataCache : IRulesConfiguration
+    internal sealed class RulesMetadataCache
     {
-        private static readonly Lazy<RulesMetadataCache> lazy = new Lazy<RulesMetadataCache>(() => new RulesMetadataCache());
+        private static IEnumerable<string> AllLanguagesAllRuleKeys { get; }
+        private static IEnumerable<string> AllLanguagesActiveRuleKeys { get; }
+        private static IDictionary<string, RuleMetadata> AllLanguagesRulesMetadata { get; }
+        private static IDictionary<string, IDictionary<string, string>> AllLanguagesRulesParameters { get; }
 
-        public static IRulesConfiguration Instance { get { return lazy.Value; } }
+        private static IDictionary<string, IRulesConfiguration> RulesByLanguage { get; }
 
-        public IEnumerable<string> AllRuleKeys { get; }
-        public IEnumerable<string> ActiveRuleKeys { get; }
-
-        public IDictionary<string, IDictionary<string, string>> RulesParameters { get; }
-
-        public IDictionary<string, RuleMetadata> RulesMetadata { get; }
+        public static IRulesConfiguration GetSettings(string cFamilyLanguage)
+        {
+            RulesByLanguage.TryGetValue(cFamilyLanguage, out var rulesConfiguration);
+            return rulesConfiguration;
+        }
 
-        private RulesMetadataCache()
+        static RulesMetadataCache()
```

Comment:
The rule files are loaded and processed in a static constructor, so it is thread-safe, and will be executed the first time the `GetSettings` method is called in code.